salt/metalk8s/orchestrate/upgrade/node.sls

```diff
@@ -0,0 +1,61 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
```

Comment:
Can't this be basically the same as, or almost the same, as `deploy_new_node.sls`?

Reply:
I.e. don't have two playbooks.

Reply:
I create one for single as we will use this for control_plane upgrade and for workload_plane_upgrade 
And then have something like that for upgrade global runner
```
└> cat salt/metalk8s/orchestrate/upgrade/init.sls 
Execute the upgrade prechecks:
  salt.runner:
    - name: state.orchestrate
    - mods:
      - metalk8s.orchestrate.upgrade.precheck

Upgrade etcd cluster:
  salt.runner:
    - name: state.orchestrate
    - mods:
      - metalk8s.orchestrate.upgrade.etcd
    - saltenv: {{ saltenv }}

Upgrade control_plane nodes:
  salt.runner:
    - name: state.orchestrate
    - mods:
      - metalk8s.orchestrate.upgrade.control_plane
    - require:
      - salt: Upgrade etcd cluster

Upgrade workload_plane nodes:
  salt.runner:
    - name: state.orchestrate
    - mods:
      - metalk8s.orchestrate.upgrade.workload_plane
    - require:
      - salt: Upgrade control_plane nodes
```
Not finished not tested of course but that my idea

Reply:
Hmh, not sure that's the best approach. If we were to do a bit more in this playbook, and basically not use any 'sub'-runners, we can more clearly decide which nodes get upgraded in which stage. As an example, there's no need to bring a control-plane node in highstate as part of the *Upgrade workload-plane nodes*, when they've been put in highstate already.